pkg/network/common/common.go

```diff
@@ -182,73 +185,42 @@ const (
 	NetworkPolicies       ResourceName = "NetworkPolicies"
 )
 
-func newEventQueue(client kcache.Getter, resourceName ResourceName, expectedType interface{}, namespace string) *EventQueue {
-	rn := strings.ToLower(string(resourceName))
-	lw := kcache.NewListWatchFromClient(client, rn, namespace, fields.Everything())
-	eventQueue := NewEventQueue(DeletionHandlingMetaNamespaceKeyFunc)
-	// Repopulate event queue every 30 mins
-	// Existing items in the event queue will have watch.Modified event type
-	go kcache.NewReflector(lw, expectedType, eventQueue, 30*time.Minute).Run(wait.NeverStop)
-	return eventQueue
-}
-
-// Run event queue for the given resource. The 'process' function is called
-// repeatedly with each available cache.Delta that describes state changes
-// to an object. If the process function returns an error queued changes
-// for that object are dropped but processing continues with the next available
-// object's cache.Deltas.  The error is logged with call stack information.
-//
-// NOTE: this function will handle DeletedFinalStateUnknown delta objects
-// automatically, which may not always be what you want since the now-deleted
-// object may be stale.
-func RunEventQueue(client kcache.Getter, resourceName ResourceName, process ProcessEventFunc) {
-	var expectedType interface{}
-
-	switch resourceName {
-	case HostSubnets:
-		expectedType = &networkapi.HostSubnet{}
-	case NetNamespaces:
-		expectedType = &networkapi.NetNamespace{}
-	case EgressNetworkPolicies:
-		expectedType = &networkapi.EgressNetworkPolicy{}
-	case NetworkPolicies:
-		expectedType = &networking.NetworkPolicy{}
-	default:
-		glog.Fatalf("Unknown resource %s during initialization of event queue", resourceName)
-	}
-
-	eventQueue := newEventQueue(client, resourceName, expectedType, metav1.NamespaceAll)
-	for {
-		eventQueue.Pop(process, expectedType)
-	}
-}
-
-// RegisterSharedInformerEventHandlers registers addOrUpdateFunc and delFunc event handlers with
-// kubernetes shared informers for the given resource name.
-func RegisterSharedInformerEventHandlers(kubeInformers kinternalinformers.SharedInformerFactory,
+// RegisterSharedInformer creates shared informer and registers addOrUpdateFunc and
+// delFunc event handlers for the given resource name.
+func RegisterSharedInformer(si SDNInformers,
 	addOrUpdateFunc func(interface{}, interface{}, watch.EventType),
 	delFunc func(interface{}), resourceName ResourceName) {
 
-	var expectedObjType interface{}
+	var objType runtime.Object
 	var informer kcache.SharedIndexInformer
 
-	internalVersion := kubeInformers.Core().InternalVersion()
-
 	switch resourceName {
 	case Nodes:
-		informer = internalVersion.Nodes().Informer()
-		expectedObjType = &kapi.Node{}
+		informer = si.KubeInformers.Core().InternalVersion().Nodes().Informer()
```

Comment:
This switch statement is against our standard conventions.  You should instead be passing in the desired informer and the obj type to this method from each of the call sites.